src/registry/number/edit.tsx

```diff
@@ -0,0 +1,221 @@
+import {NumberComponentSchema} from '@open-formulieren/types';
+import {useFormikContext} from 'formik';
+import {FormattedMessage, useIntl} from 'react-intl';
+
+import {
+  BuilderTabs,
+  ClearOnHide,
+  Description,
+  Hidden,
+  IsSensitiveData,
+  Key,
+  Label,
+  PresentationConfig,
+  Registration,
+  SimpleConditional,
+  Suffix,
+  Tooltip,
+  Translations,
+  Validate,
+  useDeriveComponentKey,
+} from '@/components/builder';
+import {Checkbox, NumberField, TabList, TabPanel, Tabs} from '@/components/formio';
+import {getErrorNames} from '@/utils/errors';
+
+import {EditFormDefinition} from '../types';
+
+/**
+ * Form to configure a Formio 'number' type component.
+ */
+const EditForm: EditFormDefinition<NumberComponentSchema> = () => {
+  const [isKeyManuallySetRef, generatedKey] = useDeriveComponentKey();
+  const {errors} = useFormikContext<NumberComponentSchema>();
+
+  const erroredFields = Object.keys(errors).length
+    ? getErrorNames<NumberComponentSchema>(errors)
+    : [];
+  // TODO: pattern match instead of just string inclusion?
```

Comment:
Copied over code, so this will be refactored at some point.

TS supports template literals and allows you to produce mapped/aliased types, e.g. it is possible to do something like

```
`[key: p as get${p}]: () => {...}`
```
where `p` is the keys of an object to make a readonly accessor.

There are then some crazy generics constructions combined with `keyof` and recursion that *I think* should make it possible to generate a typescript list/union of all possible nested key paths, e.g. `'type' | 'key' | 'validate.min' | ...` and so on, from a given component schema.

So, I think it's possible to get a type safe function where we could then use expressions like `validate.*` or something like that with a combination of regex/explicit names/arrays etc.

So long story short - I'm not entirely sure yet what the best approach here would be, but type safe checks feel like they're within reach.